src/style.js

```diff
@@ -302,43 +302,62 @@ export function* groupIndex(I, position, mark, channels) {
   }
 }
 
-// TODO avoid creating a new clip-path each time?
+// TODO Accept other types of clips (paths, urls, x, y, other marks…)?
+// https://github.com/observablehq/plot/issues/181
+export function maybeClip(clip) {
+  if (clip === true) clip = "frame";
+  else if (clip === false) clip = null;
+  else if (clip != null) clip = keyword(clip, "clip", ["frame", "sphere"]);
+  return clip;
+}
+
+function clipDefs({ownerSVGElement}) {
+  const svg = select(ownerSVGElement);
+  const defs = svg.select("defs.clip");
+  return defs.size() ? defs : svg.insert("defs", ":first-child").attr("class", "clip");
```

Comment:
Is there any benefit to a `defs` element? A `clipPath` element won’t render anything anyway and could just live at the top level. Also feels unnecessary to use the `class="clip"`.